app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.client.Firebase;
+import com.rahobbs.todo.helpers.Feedback;
+import com.rahobbs.todo.R;
+import com.rahobbs.todo.helpers.TodoItem;
+import com.rahobbs.todo.helpers.TodoLab;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * Created by rachael on 2/1/16.
+ * Fragment that contains a single to-do item detailed view
+ */
+public class TodoFragment extends Fragment {
+
+    private static final String ARG_TODO_ID = "todoId";
+    private static final String DIALOG_DATE = "DialogDate";
+    private static final int REQUEST_DATE = 0;
+
+    private TodoItem mTodo;
+    private EditText mTitle;
+    private EditText mDetails;
+    private TextView mDueDateTextField;
+    private CheckBox mCompletedCheckbox;
+    private LinearLayout mDateComponents;
+
+    public static TodoFragment newInstance(UUID todoId) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_TODO_ID, todoId);
+        TodoFragment fragment = new TodoFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Firebase.setAndroidContext(getContext());
+        UUID todoId = (UUID) getArguments().getSerializable(ARG_TODO_ID);
+        setHasOptionsMenu(true);
+
+        if (todoId == null) {
+            throw new RuntimeException("Todo ID is null.");
+        }
+
+        mTodo = TodoLab.get(getActivity()).getItem(todoId);
+        Log.v("DETAIL_CREATED", mTodo.getID().toString());
+
+        if (mTodo == null) {
+            throw new RuntimeException("No Todo item with provided ID (" + todoId.toString() + ") was found.");
+        }
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.fragment_todo_item, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_send_feedback:
+                Intent i = new Intent(Intent.ACTION_SEND);
+                Feedback fb = new Feedback();
+                fb.sendFeedback(i);
+                startActivity(Intent.createChooser(i, "Send mail..."));
+                return true;
+            case R.id.menu_item_delete_todo:
+                UUID todoId = mTodo.getID();
+                TodoLab.get(getActivity()).deleteTodoItem(todoId);
+
+                String titleToDisplay;
+                if (mTodo.getTitle() == null) {
+                    titleToDisplay = "Unnamed task";
+                } else if (mTodo.getTitle().length() > 25) {
+                    titleToDisplay = "\"" + mTodo.getTitle().trim().substring(0, 25) + "\"";
```

Comment:
String interpolation is one of the messiest most awful parts of programming. I hate it. I hate it because it is so error prone. 

Let's walk through this logic... 
1. You check if there is a title. 
2. If there isn't, you give it one
3. If there is and it's length is > 25 you just only accept the first 25 characters
4. otherwise you add spacing to it and trim it. 

First, don't use string literals always use string Ids. (i.e. `R.string.unamed_task_title`). It'll make localizing SO much easier.  Also it will remove the necessity for all the `"\"` everywhere. 

For example: 
```xml
<string name="unamed_task_title">\\ %s \\</string>
```

Then you can pass in whatever string you want the title to be. so line 106 becomes
```java
titleToDisplay = getString(R.string.unamed_task_title, mTodo.getTitle.trim())
```

Second, the behavior of only accepting the first 25 characters feels weird to me. Why not just wrap the rest? Why remove all of them?

Lastly, all this string messiness is a great place for a test! Here's how... 

1. Pull out all the code around creating this title into a static `createDeleteTodoItemToastTitle(){}`
2. Create a test file! Le'ts call it `ToastTitleTest`
3. Put in `app`>`src`>`androidTest`>`java`>`com`> `[your package]`>`ToastTitleTest`

Here's what that file could look like...

```java
import android.content.Context;
import android.test.InstrumentationTestCase;

public class TransferUtilTest extends InstrumentationTestCase {

    private Context mContext;

    @Override
    protected void setUp() throws Exception {
        mContext = getInstrumentation().getTargetContext();
    }

    public void testDeleteTodoItemToastTitle() {
        String title = ToastTitleUtils.createDeleteTodoItemToastTitle(todoItem);
        assertEquals(title, "whatever the expected title is");
    }
```

Some of that was psuedo code, I tried to be as specific as possible while writing in github ha